### PR TITLE
API access to the remaining ship class strings

### PR DESCRIPTION
**Pull request by MjnMixael**

### CODE REVIEW

File: code/scripting/api/objs/shipclass.cpp [hunk 1]
```diff
@@ -198,6 +198,97 @@ ADE_VIRTVAR(ManufacturerString, l_Shipclass, "string", "Ship class manufacturer"
...
+	if(idx < 0 || idx >= ship_info_size())
+		return ade_set_error(L, "s", "");
+
+	ship_info *sip = &Ship_info[idx];
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "Setting Length is not supported");
+	}
+
+	if(sip->manufacturer_str != NULL)
```
Comment (BMagnu) [inline code/scripting/api/objs/shipclass.cpp:217]: why are you checking against manufacturer_str  and not ship_length.
Also, please make this NULL into a nullptr

File: code/scripting/api/objs/shipclass.cpp [hunk 2]
```diff
@@ -198,6 +198,97 @@ ADE_VIRTVAR(ManufacturerString, l_Shipclass, "string", "Ship class manufacturer"
...
+	if(idx < 0 || idx >= ship_info_size())
+		return ade_set_error(L, "s", "");
+
+	ship_info *sip = &Ship_info[idx];
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "Setting Gun mounts is not supported");
+	}
+
+	if(sip->manufacturer_str != NULL)
```
Comment (BMagnu) [inline code/scripting/api/objs/shipclass.cpp:239]: same here

File: code/scripting/api/objs/shipclass.cpp [hunk 3]
```diff
@@ -198,6 +198,97 @@ ADE_VIRTVAR(ManufacturerString, l_Shipclass, "string", "Ship class manufacturer"
...
+	if(idx < 0 || idx >= ship_info_size())
+		return ade_set_error(L, "s", "");
+
+	ship_info *sip = &Ship_info[idx];
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "Setting Missile banks is not supported");
+	}
+
+	if(sip->manufacturer_str != NULL)
```
Comment (BMagnu) [inline code/scripting/api/objs/shipclass.cpp:261]: same here

**Review by BMagnu — CHANGES_REQUESTED** [review]

File: code/scripting/api/objs/shipclass.cpp (hunk 1 above)
Comment (MjnMixael) [reply code/scripting/api/objs/shipclass.cpp]: Copy Paste strikes again. Fixed

File: code/scripting/api/objs/shipclass.cpp [hunk 4]
```diff
@@ -198,6 +198,94 @@ ADE_VIRTVAR(ManufacturerString, l_Shipclass, "string", "Ship class manufacturer"
...
+	ship_info *sip = &Ship_info[idx];
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "Setting Missile banks is not supported");
+	}
+
+	char str[100];
+	sprintf(str, XSTR("%d m/s", 743), fl2i((float)sip->max_vel.xyz.z * Hud_speed_multiplier));
+
+	return ade_set_args(L, "s", "");
```
Comment (BMagnu) [inline code/scripting/api/objs/shipclass.cpp:286]: this looks like another copy paste issue. I assume this should be str instead of "", right?

Comment (MjnMixael) [reply code/scripting/api/objs/shipclass.cpp]: 👀

**Review by BMagnu — APPROVED** [review]

**Pull request merged**